app.py

```diff
@@ -105,20 +105,41 @@ def author(author_id):
     return render_template("author.html", author=author)
 
 
-@app.route("/users")
+@app.route("/users", methods=["GET", "POST"])
 def users():
     """Show all of the users and basic user details"""
-    return render_template("users.html", users=users_info)
+
+    if request.method == "POST":
+        if request.form.get("user_level"):
+            programming_level = request.form["user_level"]
+            users = db_session.query(User).filter(User.user_level == programming_level)
+
+        elif request.form.get("last_name"):
+            user_last_name = request.form["last_name"]
+            users = db_session.query(User).filter(User.last_name == user_last_name)
+
+    elif request.method == "GET" and request.args:
+        if request.args.get("sort"):
+            sort_by = request.args.get("sort")
+            users = db_session.query(User).order_by(sort_by).all()
```

Comment:
You don't need this `.all()` since you're ultimately iterating through this result object in the view. Notice you don't call `all()` in the previous queries in this view function. Not a big deal though.